flutter/flutter-provider/state-management-in-flutter-using-provider-Part2.md

```diff
@@ -0,0 +1,198 @@
+# State Management in Flutter using Provider: Part 2
+
+In the previous [article](https://suresh.hashnode.dev/state-management-in-flutter-using-provider-part-1-cke8vk1cb00iny0s11lu933ch),
+we have discussed the problem of using `setState` to update the app's state in the moderate or complex app.
+
+## What we are going to learn in this article
+
+1. What is Provider, Why we use it?
+2. What are the types of Provider?
+3. How to use Provider in our code?
+4. What is Consumer & Selector?
+  
+### 1. What is Provider, Why we use it
+
+`Provider` is one of the most popular state management solution, created by  [Remi Rousselet](https://github.com/rrousselGit/provider), can be used in flutter app development.
+
+On the beginner stage, we use `setState` to update the app's state, because it is easy, and if we try to use some other moderate or say a complex technique like `flutter_bloc` or `Mobx`, this is not easy to switch from `setState` to these state management technique 😓
```

Comment:
Break sentence after "it is easy". Next line can look like "And switching from `setState` to other complex techniques like `flutter_bloc` or `Mobx` is not that easy to follow."